bootstrap.py

```diff
@@ -383,13 +414,35 @@ def install_puppet_agent():
     call_yum("install", "puppet")
     enable_service("puppet")
 
-    puppet_conf = open('/etc/puppet/puppet.conf', 'wb')
-    puppet_conf.write("""
+    puppet_major_version = get_puppet_version()
+    if puppet_major_version == 3:
+        puppet_conf_file = '/etc/puppet/puppet.conf'
+    elif puppet_major_version in [4, 5]:
+        puppet_conf_file = '/etc/puppetlabs/puppet/puppet.conf'
+    else:
+        print_error("Unsupported puppet version")
+        sys.exit(1)
+    puppet_conf = open(puppet_conf_file, 'wb')
+    if puppet_major_version == 3:
+        puppet_conf.write("""
 [main]
 vardir = /var/lib/puppet
 logdir = /var/log/puppet
 rundir = /var/run/puppet
 ssldir = $vardir/ssl
+""")
+    elif puppet_major_version in [4, 5]:
+        puppet_conf.write("""
+[main]
+vardir = /opt/puppetlabs/puppet/cache
+logdir = /var/log/puppetlabs/puppet
+rundir = /var/run/puppet
+ssldir = /etc/puppetlabs/puppet/ssl
+""")
+    else:
```

Comment:
that code is unreached, we already verified that it's version 3, 4 or 5 when looking for the conf path